src/logic/hooks/useAsyncValue.ts

```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect } from 'react'
+
+// Initial dependencies must be ordered according to the args of asyncCall
+const useAsyncValue = <T>(
+  asyncCall: (...deps: unknown[]) => Promise<T>,
+  getDeps: () => Parameters<typeof asyncCall>,
```

Comment:
If we strictly type `asyncCall`, `getDeps()` will thorw if there are more than the arguments for `asyncCall`. As such, it's typed as `unknown[]`